test/dist_autograd_test.py

```diff
@@ -160,3 +268,11 @@ def test_rpc_complex_args(self):
                     self.assertEqual(tensors[i], next_funcs[i][0].variable)
                 else:
                     self.assertIsNone(next_funcs[i][0])
+
+    @dist_init
+    def test_nested_contex(self):
+        with self.assertRaises(RuntimeError):
```

Comment:
Will the `RuntimeError` also come with a string message? Let's also verify it is the exact RuntimeError we are expecting. 

Q: is this just a temporary limitation that we do not support nested context or will it always be in that way?

Reply:
it is always like this, it did not work for nested context right now, as current_context_id will be cleared when inner context exits, I'm just adding checks to make it more explicit